use linker generated section brackets

MartinNowak opened the pull request:
- rename sections to "minfo" and "deh"
- use hidden undefined __start/stop_minfo/deh
  to get start and end of sections
- linker will define those symbols
  (see http://www.airs.com/blog/archives/56)

MartinNowak commented:
Requires https://github.com/D-Programming-Language/druntime/pull/1379

WalterBright commented on src/backend/elfobj.c:
```diff
@@ -463,13 +462,15 @@ static IDXSTR elf_addmangled(Symbol *s)
  *      bind    =       symbol binding
- *      segidx  =       segment index for segment where symbol is defined
+ *      sec     =       index of section where symbol is defined
+ *      vsblty  =       visibility of symbol
```
I hate to complain about names, but why not just call it `visibility`?

Also, what is visibility of names? You could add `STV_xxxx` to the explanatory comment.

MartinNowak replied:
done

WalterBright commented:
Nice. I wish I'd known about that linker feature sooner!

MartinNowak committed:
use linker generated section brackets

- rename sections to "minfo" and "deh"
- use hidden undefined __start/stop_minfo/deh
  to get start and end of sections
- linker will define those symbols
  (see http://www.airs.com/blog/archives/56)
- drop _Dmodule_ref initializers

MartinNowak commented:
I want to use this to add a separate section for `ptrdata` so that the GC needs to scan less roots.
[Issue 14472 – add separate ptr data section](https://issues.dlang.org/show_bug.cgi?id=14472), [ptr data section on Active | Trello](https://trello.com/c/Dsdxd1r3/56-ptr-data-section).
But it seems I'd have to change `dt` generation quite a lot to achieve this.

MartinNowak commented:
Maybe we can combine that with [Issue 14816 – improve dt_t data type for faster appending (tail list or array)](https://issues.dlang.org/show_bug.cgi?id=14816).

rainers commented:
> I want to use this to add a separate section for ptrdata so that the GC needs to scan less roots.

Here is a commit with a similar functionality for the precise GC presented at DConf2013: https://github.com/rainers/dmd/commit/e6a3d1778ad2969fc7bcb378a084b8d865b015c7

It wrote (address,TypeInfo) pairs, but just emitting pointers to pointers sounds simpler.
